springcloud-streams/src/main/java/org/axonframework/extensions/springcloud/eventhandling/SpringStreamMessageSource.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2010-2021. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.springcloud.eventhandling;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.Registration;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.extensions.springcloud.eventhandling.converter.SpringMessageConverter;
+import org.axonframework.messaging.SubscribableMessageSource;
+import org.springframework.integration.handler.AbstractMessageProducingHandler;
+import org.springframework.messaging.Message;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+import static java.util.Collections.singletonList;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Initialize an SpringStreamMessageSource instance that sends all incoming spring {@link Message} to the given
+ * {@link EventBus}. It is still possible for other Event Processors to subscribe to this MessageChannelAdapter.
+ *
+ * @author Mehdi Chitforoosh
+ * @since 4.5
+ */
+public class SpringStreamMessageSource extends AbstractMessageProducingHandler implements SubscribableMessageSource<EventMessage<?>> {
+
+    private final CopyOnWriteArrayList<Consumer<List<? extends EventMessage<?>>>> messageProcessors = new CopyOnWriteArrayList<>();
```

Comment:
I think it's better to call these `eventProcessors` as they are consumers of the `EventMessage` type.